### PR TITLE
Default to older file header unless there is a .bmp (Follow-up on #11)

### PR DESCRIPTION
**Pull request by adriweb**
Updated/Fixed version of [ndless-nspire/Luna@`6f9fa1b` (#11)](https://github.com/ndless-nspire/Luna/pull/11/commits/6f9fa1b25b25a4a40dcc140a2ab795cbd4aae4b7)

Ugly but seems to work?

### CODE REVIEW

File: minizip-1.1/zip.c [hunk 1]
```diff
@@ -977,11 +980,25 @@ int Write_LocalFileHeader(zip64_internal* zi, const char* filename, uInt size_ex
...
 	  {
-	  	err = zip64local_putValue(&zi->z_filefunc,zi->filestream, LOCALHEADERMAGIC2, 4);
+	  	err = zip64local_putValue(&zi->z_filefunc,zi->filestream, LOCALHEADERMAGIC2, 2);
 	  }
 	  if (err==ZIP_OK)
 	  {
-	  	err = zip64local_putValue(&zi->z_filefunc,zi->filestream, LOCALHEADERMAGIC3, 2);
+	  	char buf[5] = {0};
+	  	snprintf(buf, 5, "%02X%02X", tiapimajor, tiapiminor);
+	  	err = zip64local_putValue(&zi->z_filefunc,zi->filestream, buf[0], 1);
```
Comment (Vogtinator) [inline minizip-1.1/zip.c:989]:
```
ZWRITE64(zi->z_filefun, zi->filestream, buf, 4);
```

**Comment by Vogtinator** [comment]
IMO the version can be treated as a single `uint16_t` with the MSB for the major and the LSB for the minor version. That's how it looks in the file anyway.

File: minizip-1.1/zip.c (hunk 1 above)
Comment (adriweb) [reply minizip-1.1/zip.c:989]: Oh, I had tried the 4 bytes at the time and didn't work, but I guess with `ZWRITE64` it might... let me see.

Comment (adriweb) [reply minizip-1.1/zip.c]: seems to work fine

**Comment by adriweb** [comment]
> IMO the version can be treated as a single `uint16_t` with the MSB for the major and the LSB for the minor version. That's how it looks in the file anyway.

Ah well, probably, but it doesn't matter too much... does it? :P I guess you have edit rights on the PR files anyway 👀

**Comment by adriweb** [comment]
nice

**Comment by Vogtinator** [comment]
Feel free to test again with the changes. Here it looks ok.

**Comment by adriweb** [comment]
Well, converting a .lua file worked fine (and resulted in a 0500 as expected), but converting a simple .bmp (`./luna test.bmp test.tns` ?) generated a 0700 but triggered an error while trying to open it in TINCS...

**Comment by Vogtinator** [comment]
> I feel like just having a bmp as input isn't enough though, so hopefully that's why :) Don't we have some test xmls somewhere to try this :D

The ndless 3.9.1 installer uses two BMPs

**Comment by adriweb** [comment]
Found something else, works fine in TINCS :)

**Pull request merged**